compiler/src/dotty/tools/dotc/transform/CheckPhantomCast.scala

```diff
@@ -0,0 +1,46 @@
+package dotty.tools.dotc
+package transform
+
+import core._
+import dotty.tools.dotc.transform.TreeTransforms.{MiniPhaseTransform, TransformerInfo}
+import Types._
+import Contexts.Context
+import Symbols._
+import Decorators._
+import dotty.tools.dotc.ast.Trees._
+import dotty.tools.dotc.ast.tpd
+
+
+/** A no-op transform that checks whether the compiled sources have no Phantom types in casts */
+class CheckPhantomCast extends MiniPhaseTransform { thisTransformer =>
+
+  override def phaseName = "checkPhantomCast"
```

Comment:
add a post-condition that those casts don't reappear so that Ycheck will enforce it?